### PR TITLE
Adding the intial lexer

### PR DESCRIPTION
**Pull request by AbooMinister25**
This code has the initial code for the lexer, it supports a variety of single and double character tokens, integers, strings, bools, identifiers, and keywords. It also has the initial style changes from camelCase to snake_case.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -24,7 +26,21 @@ static void repl() {
...
+                if (token.kind == T_ERROR) {
+                    printf("Error: %s\n", token.start);
+                    continue;
+                }
+                
+                print_token(&token);
+            }
         }
     }
 }
```
Comment (dawnofmidnight) [inline src/main.c:46]: I can't comment below this line, but might we want some intro to the repl? Something like this, maybe?
```
Meow 0.1.0. For help, type `:help`
> 
```

File: src/tokens.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    T_FLOAT,
+
+    // Identifier
+    T_IDENT,
+
+    // Single character tokens
+    T_LPAREN,
+    T_RPAREN,
+    T_LBRACE,
+    T_RBRACE,
```
Comment (dawnofmidnight) [inline src/tokens.h:17]: We also want `LBracket` and `RBracket`, right?

File: src/lexer.c [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+        case '.':
+            return create_token(lexer, T_DOT);
+        case '+':
+            return create_token(lexer, T_PLUS);
+        case '-':
+            return create_token(lexer, T_MINUS);
+        case '*':
+            return create_token(lexer, T_STAR);
+        case '/':
+            return create_token(lexer, T_SLASH);
```
Comment (dawnofmidnight) [inline src/lexer.c:261]: When testing, I noticed that the lexer hangs on any slash. Also, are we not including comments in this initial iteration?

**Review by dawnofmidnight — CHANGES_REQUESTED** [review]
Other than the following comments, this looks great! I'm probably forgetting some really important things somewhere, but I'll review again later on.

File: src/lexer.c (hunk 3 above)
Comment (AbooMinister25) [reply src/lexer.c:261]: There are comments, they're handled in the skip_whitespace function, and are only handled if two successive slashes are found

File: src/main.c (hunk 1 above)
Comment (AbooMinister25) [reply src/main.c:46]: Sure, we can do that

File: src/tokens.h (hunk 2 above)
Comment (AbooMinister25) [reply src/tokens.h:17]: Oop, I'll get that down

File: src/lexer.c (hunk 3 above)
Comment (dawnofmidnight) [reply src/lexer.c:261]: Ah, ok

Comment (dawnofmidnight) [reply src/lexer.c:261]: Although the hanging thing, I don't know. It might just be a repl thing, but the thing just died there.

Comment (AbooMinister25) [reply src/lexer.c:261]: Yeah, I'll look into that tomorrow

File: src/vm.h [hunk 4]
```diff
@@ -15,9 +15,9 @@ typedef struct {
     Value *stackTop; // Pointer to the top value in the VM's stack.
 } VM;
 
-void initVM(VM *vm); // Function for initializing the Meow VM.
+void init_vm(VM *vm); // Function for initializing the Meow VM.
 
-void freeVM(VM *vm); // Function that frees the Lite VM from memory.
+void free_vm(VM *vm); // Function that frees the Lite VM from memory.
```
Comment (doublevcodes) [inline src/vm.h:20]: Uhh... Lite?

**Review by doublevcodes — APPROVED** [review]
Lookin' decent

**Review by dawnofmidnight — APPROVED** [review]

**Pull request merged**